gst/tensor_decoder/tensordec.c

```diff
@@ -688,6 +688,56 @@ gst_tensordec_copy_buffer (GstTensorDec * self,
 static void
 gst_tensordec_update_top_label_index (GstTensorDec * self,
     guint8 * scores, guint len)
+{
+  gint i;
+  gint index = -1;
+  guint8 max_score = 0;
+
+  /** -1 if failed to get max score index */
+  self->tensordec_image_label.new_label_index = -1;
+
+  g_return_if_fail (scores != NULL);
+  g_return_if_fail (len ==
+      self->tensordec_image_label.labeling_info.total_labels);
+
+  for (i = 0; i < len; i++) {
+    if (scores[i] > 0 && scores[i] > max_score) {
+      index = i;
+      max_score = scores[i];
+    }
+  }
+
+  self->tensordec_image_label.new_label_index = index;
```

Comment:
Why don't you return the index with the max-score?
Why do we have such values in the class (self)? It's a volatile value (effective for only one frame) that we do not need to save.

Reply:
I will change image labeling data structure as not using values in class